lib/fireblocks/api/request.rb

```diff
@@ -41,21 +41,24 @@ def put(body)
       valid_response!(send_request(req), request: req)
     end
 
-    def post(body)
+    def post(body, headers)
       req = Net::HTTP::Post.new(uri)
-      request_headers(body).each { |rk, rv| req[rk] = rv }
+      request_headers(body, headers).each { |rk, rv| req[rk] = rv }
+      req['Idempotency-Key'] = idempotency_key if idempotency_key
+      @idempotency_key = idempotency_key
       req.body = body.to_json
 
       valid_response!(send_request(req), request: req)
     end
 
     private
 
-    def request_headers(body)
+    def request_headers(body, headers = {})
       {
         'X-API-Key' => Fireblocks.configuration.api_key,
         'Authorization' => "Bearer #{token(body)}",
-        'Content-Type' => 'application/json'
+        'Content-Type' => 'application/json',
+        'Idempotency-Key' => headers['Idempotency-Key']
```

Comment:
this is going to be sending an empty header value if this is not defined. Is that going to cause issues?